src/extension/src/scripts/run-tests.ts

```diff
@@ -0,0 +1,150 @@
+let child_process = require("child_process");
+const path = require("path");
+const fs = require("fs");
+
+const testFolder = path.join(__dirname, "..", "..", "src", "template_test");
+console.log(testFolder);
+let files: string[] = [];
+
+//Recursively delete all generated projects in template_test
+let deleteFolderRecursive = function(pathname: string) {
+  let files = [];
+  if (fs.existsSync(pathname)) {
```

Comment:
Is it possible to call the script to generate templates if the pathname is not found? That way, we only need to run 1 command, i.e. `yarn run-tests`, to run the tests (instead of `yarn generate` and then `yarn run-tests`)

Reply:
You raise a valid point but there's actually a couple of reasons why I didn't follow that approach. 

The generate-test script is for populating template_test folder after it's executed (which is empty in the beginning of every test) and its additional purpose it to make sure that coretsapi also has no problems because otherwise, it would fail to properly generate all the projects with its corresponding pages. 

The template_test folder is emptied after run-tests finishes so we don't have unnecessary files that make the build fail. 

When you run ./build if the generated projects are there, it will fail. So that's also why it is deleted afterwards.